### PR TITLE
Feature/xcode

### PR DESCRIPTION
**Pull request by mnbittorrent**
I made some changes no needy so that it mostly will accept Xcode-friendly platform names. There were inconsistencies growing between the Xcode platforms, so it was useful to derive the Xcode platforms from an `XcodePlatform` base class.

I also added a "Mac OS X" platform so that we can properly pass in the targeted macosx version. Darwin hosts will pick this `MacOSXPlatform` over the default `HostPlatform`.

I was having trouble getting needy to rebuild things based off of arguments that affect the build, so I added `configuration_hash()` to help with that. It's pretty straightforward.

The `--force-build` command was also very useful for testing.

### CODE REVIEW

File: needy/generators/jamfile.py [hunk 1]
```diff
@@ -61,7 +79,16 @@ def generate(self, needy):
...
+""" % (
+    os.path.abspath(sys.argv[0]),
+    os.path.dirname(needy.path()),
+    needy.path(),
+    target_args['ios'],
+    target_args['iossimulator'],
+    target_args['tvos'],
+    target_args['tvossimulator'],
+    target_args['osx'],
+    needy.parameters().satisfy_args)
```
Comment (ccbrown) [inline needy/generators/jamfile.py]: feel free to switch to named arguments here if you'd like

File: needy/library.py [hunk 2]
```diff
@@ -249,4 +258,8 @@ def __configuration_hash(self, target):
 
         hash.update(json.dumps(self.project_configuration(target), sort_keys=True))
 
-        return hash.digest()
+        platform_configuration_hash = target.platform.configuration_hash(target.architecture)
+        if platform_configuration_hash:
+            hash.update(platform_configuration_hash)
+
+        return hash.hexdigest()
```
Comment (ccbrown) [inline needy/library.py]: i have a slight preference for only converting to hex in the serialization step

**Comment by ccbrown** [comment]
i think this looks good. i'll test and merge it tonight

**Comment by ccbrown** [comment]
seems solid, but it did produce one unexpected side-effect: when building for the 'host' platform, conditionals still only matched 'darwin' on osx. so i'm adding an `if sys.platform == 'darwin': values.append('osx')` to the conditional evaluation.

i also think it's a little weird that the 'host' platform isn't always the 'osx' platform on osx... i don't actually think i want to combine them though, because they are different things... i might rename 'host' to 'generic' and make 'host' act as an alias for either the 'generic' or 'osx' platform, depending on your host os

**Comment by mnbittorrent** [comment]
Sounds good.

I was playing around with treating osx as host, but that left no fallback
for the case where someone might want to take advantage of specifying CC.
Most of the time you'll want to just use osx as host, but maybe not always.

I like the idea of 'generic' or maybe 'default' for that so it makes sense.
You might also rename 'host' to 'clang' since that's more specifically what
it uses. I would drop in a 'gcc' one too just for good measure if you like
that idea.

**Pull request merged**